packages/plugin-mode/src/Mode.ts

```diff
@@ -49,50 +49,21 @@ export class Mode<T extends ModeConfig = ModeConfig> extends JWPlugin<T> {
     setCurrent(modeIdentifier: ModeIdentifier): void {
         this._currentMode = modeIdentifier;
     }
-    /**
-     * Return true if the given node's content is editable.
-     *
-     * @param node
-     */
-    isEditable(node: VNode): boolean {
-        const hierarchy = [...node.ancestors(), node];
-        const mode = this._definitions[this._currentMode];
-        const ruleEntries = mode.rules
-            .filter(rule => typeof rule.editable !== 'undefined')
-            .map(rule => {
-                const entry: ContextualEntry<boolean> = {
-                    selector: rule.selector,
-                    data: rule.editable,
-                };
-                return entry;
-            });
-        const result = this.editor.contextManager.getRankedMatches(hierarchy, ruleEntries);
-        if (typeof result[0] === 'undefined') {
-            return node.editable;
-        } else {
-            return result[0].entry.data;
-        }
-    }
-    /**
-     * Return true if the given node is breakable.
-     *
-     * @param node
-     */
-    isBreakable(node: VNode): boolean {
+    is(node: VNode, property: 'editable' | 'breakable'): boolean {
```

Comment:
Au début quand j'ai vu ce truc je me suis dit "oulalalalah il est reparti avec ses paramètres en string". Puis j'ai vu qu'en fait ça se "combine" pas trop mal avec le fait que c'est le nom d'une property sur les rules. Du coup je me demande si on pourrait pas leverage ce concept à fond et du coup retirer la property `data` que je trouve affreuse par du coup le nom de la property et la rule qui du coup serait typée du style :
```ts
export type Rule {
    selector: Predicate[];
    [key: string]: boolean
}
```
Le problème c'est qu'un type comme ça TypeScript le refuse parce qu'il va dire que `selector` ne matche pas l'index signature puisque lui c'est pas un boolean. Je sais plus comment on fait pour faire ça mais je pense qu'il y a moyen. Genre ça p-e (names are wip):
```ts
type Property: string;
type Ruling: Record<Property, boolean>;
type Rule = Omit<Ruling, 'selector'> & {
    selector: Predicate[];
}
```
Et du coup le type du paramètre de `is` ce serait `Property`. Après je suis pas sûr que ça marche mon truc avec `Omit`, mais j'imagine qu'il y a bien une façon de typer un record en disant qu'il y a *une* propriété particulière qui n'est pas du même type.

Reply:
Je suis suis contre un string sur le `Rule`. Le `data` est fait exprès pour être générique.
D’ailleurs la généricité est déjà utilisée 2x avec `getRankedMatches` 

Je suis aussi contre les paramètres en string. C'est nul pour la recherche dans le code.

Reply:
Si c'est pour le grep, tu fais un `enum` et c'est bon...
```ts
export enum ModeProperty {
    EDITABLE = 'editable',
    BREAKABLE = 'breakable',
}
```
Quant à l'argument de l'évolution future, c'est probablement une bonne idée mais c'est très lointain et très abstrait actuellement donc je serais d'avis de ne pas trop s'en préoccuper pour le moment, chaque chose en son temps. Je préfère que ça prenne deux jours de plus dans deux ans qu'un jour de plus maintenant.

Reply:
D'accord mais on ne parle pas d'une journée juste pour une clé (qui existe déjà et qui fait bien le taff) ^^.

Reply:
Si j'ai bien compris @Goaman n'aime pas ma suggestion de "généraliser" le paramètre de la méthode `is` à tout le système de mode. Par contre je n'ai pas l'impression qu'il ait donné son avis sur le fait d'avoir un string dans la méthode `is` elle-même. Mais bon en réalité je pense que je connais son avis. cc @Zinston si tu veux défendre ton `is` ? Parce que j'ai pas trop compris votre histoire de deux jours plus tard vs un jour maintenant ou je sais pas quoi... :p